local-volume/provisioner/pkg/common/common.go

```diff
@@ -49,6 +50,15 @@ const (
 
 	// EventVolumeFailedDelete copied from k8s.io/kubernetes/pkg/controller/volume/events
 	EventVolumeFailedDelete = "VolumeFailedDelete"
+
+	// BlockCleanupFastRest does a quick reset of any previous file system information.
+	BlockCleanupFastRest = "quick-reset"
```

Comment:
I think it would be good to have an option for "blkdiscard" as a quick zeroing (but not secure) method for SSDs.  In general, for SSDs, writing zeros or random bytes at the block layer is not effective.  It's much better and faster to use the device commands available.  Some SSDs may support "secure erase" using hdparm as well.

Reply:
I assuming that you would like to rename quick-reset to blkdiscard, will do so, Regarding zero-ing, yes there is an option to do that (and also random), but its not a quick process. It will run for a while for larger disks and as you mentioned, less effective for ssd. hdparm is going to be supported, I plan to provide a custom script which would script that.

Reply:
No I mean an additional cleanup method for blkdiscard.  We can still keep quick-reset for destroying the filesystem.

Reply:
So that option is currently named "zero". I can rename that to "blkdiscard".

Here are the ones currently supported:
	// BlockCleanupFastRest does a quick reset of any previous file system information.
	BlockCleanupFastRest = "quick-reset"
	// BlockCleanupZero does zeroing out of all content in block device
	BlockCleanupZero = "zero"  ==> "blkdiscard"
	// BlockCleanupRandom Overwrite with random values.
	BlockCleanupRandom = "random"
	// BlockCleanupCustom Use custom script to do cleanup.
	BlockCleanupCustom = "custom"

Reply:
I'm only aware of using blkdiscard for SSDs.  Do HDDs support it?  I'm wondering if we need two separate zero routines.

Reply:
Ah OK. I apologize, I thought you meant the choice of names for the different operations. You were referring to the utility blkdiscard. Yes, that could be added as an option here. The other possibility is to make it one of the currently available custom scripts. I imagine we might get other requests like "scrub", "wipe", etc and those could follow a similar pattern. Let me know what you think.

Reply:
Or should we have no built-in options and just have everything be a plugin via script?

Reply:
I like that idea.

Reply:
done.